--- a/bytecode.pl
+++ b/bytecode.pl
@@ -1044,4 +1044,5 @@ __END__
 165 18	op_savefree	PL_op->op_savefree		U8
 166 18	op_static	PL_op->op_static		U8
 167 19.003 op_folded	PL_op->op_folded		U8
-168 21.002 op_lastsib	PL_op->op_lastsib		U8
+168 21.002-21.010 op_lastsib	PL_op->op_lastsib		U8
+169 21.011 op_moresib	PL_op->op_moresib		U8
--- a/lib/B/Bytecode.pm
+++ b/lib/B/Bytecode.pm
@@ -846,9 +846,12 @@ sub B::OP::bsave_thin {
     if ($] >= 5.019002 and $op->can('folded')) {
       asm "op_folded", $op->folded if $op->folded;
     }
-    if ($] >= 5.021002 and $op->can('lastsib')) {
+    if ($] >= 5.021002 and $[ < 5.021011 and $op->can('lastsib')) {
       asm "op_lastsib", $op->lastsib if $op->lastsib;
     }
+    if ($] >= 5.021011 and $op->can('moresib')) {
+      asm "op_moresib", $op->moresib if $op->moresib;
+    }
   }
 }
