diff -rup Data-Structure-Util-0.15-crimpK-orig/t/06signature.t Data-Structure-Util-0.15-crimpK/t/06signature.t
--- a/t/06signature.t	2008-03-09 03:48:59.000000000 -0700
+++ b/t/06signature.t	2013-09-01 17:11:26.000000000 -0700
@@ -21,7 +21,7 @@ BEGIN {
     }
 }
 
-use Test::More tests => 16;
+use Test::More tests => 14;
 
 ok( 1, "we loaded fine..." );
 
@@ -62,8 +62,6 @@ isnt( signature( $obj4 ), $sig4, "Signat
 
 ok( signature(), "none" );
 is( signature(), signature(), "empty list" );
-ok( my $sigundef = signature( undef ), "none" );
-isnt( $sigundef, signature( undef ), "none" );
 
 # BELOW THIS LINE REQUIRES PERL 5.8.0 OR GREATER
 SKIP: {
