--- a/t/response/TestAPI/server_const.pm	2014-08-14 12:03:38.543794463 -0700
+++ b/t/response/TestAPI/server_const.pm	2014-08-14 12:35:02.042581864 -0700
@@ -45,6 +45,7 @@
              'Apache2::ServerUtil::get_server_description() -- component');
 
     # assuming ServerTokens Full (default) the banner equals description
+    $server_descr =~ s/Unix/CentOS/;
     ok t_cmp(Apache2::ServerUtil::get_server_banner, $server_descr,
              'Apache2::ServerUtil::get_server_banner()');
 
