--- a/Apache-Reload/t/lib/Apache2/TestReload.pm 2014-08-19 13:36:49.324352146 -0700
+++ b/Apache-Reload/t/lib/Apache2/TestReload.pm 2014-08-19 13:36:51.357351017 -0700
@@ -15,7 +15,7 @@
 sub handler {
     my $r = shift;
     $pass++;
-    if ($r->args eq 'last') {
+    if ($r->args || '' eq 'last') {
         Apache2::Reload->unregister_module($package);
         ModPerl::Util::unload_package($package);
         $pass = 0;
