--- a/t/08-recurse.t	2016-03-14 14:03:56.877526835 -0700
+++ bt/08-recurse.t	2016-03-14 14:04:03.853634598 -0700
@@ -8,7 +8,7 @@
 use Net::DNS::Resolver::Recurse;
 
 
-exit( plan skip_all => 'Online tests disabled.' ) if -e 't/online.disabled';
+exit( plan skip_all => 'Online tests disabled.' ); # if -e 't/online.disabled';
 exit( plan skip_all => 'Online tests disabled.' ) unless -e 't/online.enabled';
 
 
