--- a/t/SOAP/Transport/HTTP/CGI/test_server.pl
+++ b/t/SOAP/Transport/HTTP/CGI/test_server.pl
@@ -2,7 +2,7 @@
 
 use strict;
 use warnings;
-
+use utf8;
 use SOAP::Lite;
 use SOAP::Transport::HTTP;
 
@@ -17,7 +17,7 @@
 
 use Encode;
 #    return SOAP::Data->name('testResult')->value('Überall')->type('string');
-    return SOAP::Data->name('testResult')->value(Encode::encode_utf('Überall'))->type('string');
+    return SOAP::Data->name('testResult')->value(Encode::encode_utf8('Überall'))->type('string');
 }
 
-1;
\ No newline at end of file
+1;
